concent_api/middleman/queue_operations.py

```diff
@@ -0,0 +1,186 @@
+                )
+            )
+            continue
+        logger.info(f"Received message from Concent: request ID = {frame.request_id}, connection ID = {connection_id}")
+        item = RequestQueueItem(
+            connection_id,
+            frame.request_id,
+            frame.payload,
+            now().timestamp()
+        )
+        await request_queue.put(item)
+
+
+async def request_consumer(
+    request_queue: Queue,
+    response_queue_pool: Dict[int, Queue],
+    message_tracker: Dict[int, MessageTrackerItem],
+    signing_service_writer: StreamWriter
+):
+    message_tracker = message_tracker  # type: OrderedDict[int, MessageTrackerItem]
+    signing_service_request_id = 0
+    while True:
+        item = await request_queue.get()  # type: RequestQueueItem
+        assert isinstance(item, RequestQueueItem)
+        if item.connection_id not in response_queue_pool:
+            logger.info(f"No matching queue for connection id: {item.connection_id}")
+            request_queue.task_done()
+            continue
+        signing_service_request_id = (signing_service_request_id + 1) % CONNECTION_COUNTER_LIMIT
+        message_tracker[signing_service_request_id] = MessageTrackerItem(
+            item.concent_request_id,
+            item.connection_id,
+            item.message,
+            now().timestamp()
+        )
+        frame = create_middleman_protocol_message(PayloadType.GOLEM_MESSAGE, item.message, signing_service_request_id)
+        await send_over_stream_async(frame, signing_service_writer, settings.CONCENT_PRIVATE_KEY)
+        request_queue.task_done()
+
+
+async def response_producer(
+    response_queue_pool: Dict[int, Queue],
+    signing_service_reader: StreamReader,
+    message_tracker: Dict[int, MessageTrackerItem],
+):
+    message_tracker = message_tracker  # type: OrderedDict[int, MessageTrackerItem]
+    while True:
+        try:
+            frame = await handle_frame_receive_async(signing_service_reader, settings.SIGNING_SERVICE_PUBLIC_KEY)
+        except IncompleteReadError:
+            logger.info(f"Signing Service has closed the connection. Ending 'response_producer' coroutine")
+            break
+        except (FrameInvalidMiddlemanProtocolError,
+                BrokenEscapingInFrameMiddlemanProtocolError,
+                SignatureInvalidMiddlemanProtocolError,
+                PayloadTypeInvalidMiddlemanProtocolError) as exception:
+            logger.info(f"Received invalid message: error code={map_exception_to_error_code(exception)}")
+            print(exception)
+            continue
+        logger.info(
+            f"Received message from Signing Service: request ID = {frame.request_id}"
+        )
+        if frame.request_id not in message_tracker:
+            logger.info(f"There is no entry in Message Tracker for request ID = {frame.request_id}, skipping...")
+            continue
+        current_track = message_tracker[frame.request_id]  # type: MessageTrackerItem
```

Comment:
I think this part is super compact, it won't hurt to add some blank lines between pieces.